### DISCUSSION TITLE
Give Tasks the ability to continue processing if a step is unsuccessful

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8121] [Ideas]
> Created by: Brandon Kelly (brandon@pixelandtonic.com) on 2015/03/06 03:54:45 +0000
> Votes at time of UserVoice import: 29

-----------------------------------

It would be nice if a Task's runStep() could report that the step failed, but have the TasksService continue running the remaining steps.

Perhaps with a "$continueOnError" property on the task.

**Comment by angrybrad** [comment]
> Posted by Johan Strömqvist (naboovalley@gmail.com) on 2015/03/11 06:32:09 +0000
-----------------------------------

A little more intuitive idea would to have some sort of timeout limit. So if the timestamp of the running task is older then X minutes/hours/days just clear the task OR mark it as failed and move along with the other remaining tasks.

**Comment by angrybrad** [comment]
> Posted by Florian (fw@demodern.de) on 2015/03/13 06:38:29 +0000
-----------------------------------

Yea most definitely...running remaining tasks....as well as better task management (rerun tasks, timeout on stalled tasks and auto-rerun) would be great...we keep getting into errors with that...on some servers

**Comment by angrybrad** [comment]
> Posted by Johannes Lamers (johanneslamers@gmail.com) on 2015/03/10 17:03:25 +0000
-----------------------------------

Plus +10 for this!

Hate it to start Sequal pro and set the craft_tasks table status row to &quot;error&quot;.

Better to have a button in the CP to rerun of continue the task.

**Comment by peteeveleigh** [comment]
+ several million (hours of debugging stuck tasks)
